src/annotator/guest.js

```diff
@@ -149,16 +148,19 @@ export default class Guest {
 
     // Setup the document type-specific integration consisting of metadata extraction,
     // anchoring module and logic to respond to activity (eg. scrolling) in the page.
-    //
-    // In future the plan is to put these behind a common interface so that the
-    // `Guest` class doesn't need different code paths depending on the document type.
-
-    // nb. The `anchoring` field defaults to HTML anchoring and in PDFs is replaced
-    // by `PDFIntegration` below.
-    this.anchoring = anchoring;
-    this.documentMeta = new DocumentMeta();
     if (config.documentType === 'pdf') {
-      this.pdfIntegration = new PDFIntegration(this);
+      this.integration = new PDFIntegration(this);
+    } else {
+      const documentMeta = new DocumentMeta();
+      this.integration = {
+        anchor: htmlAnchoring.anchor,
+        contentContainer: () => this.element,
+        describe: htmlAnchoring.describe,
+        destroy: () => {},
+        fitSideBySide: () => false,
+        getMetadata: () => Promise.resolve(documentMeta.getDocumentMetadata()),
+        uri: () => Promise.resolve(documentMeta.uri()),
+      };
```

Comment:
The HTML implementation of `integration` is currently defined inline in the `Guest` class. I plan to extract it into a separate `HTMLIntegration` class in future for consistency with `PDFIntegration`.

Reply:
👍🏼

Reply:
Please, would it be possible for you to create the `Integration` interface and use `@implements {Integration}`on the `PDFIntegration` class and temporary `@type {Integration}` on `this.integration` declaration?

Reply:
Note for posterity: This change was implemented in the follow-up PR https://github.com/hypothesis/client/pull/3202